testsuite/features/step_definitions/common_steps.rb

```diff
@@ -89,9 +89,13 @@
   step %(I should see a "#{ram_mb}" text)
 end
 
-When(/^I check the MAC address value$/) do
-  get_mac_address = 'cat /sys/class/net/eth0/address'
-  mac_address, _local, _remote, _code = $client.test_and_store_results_together(get_mac_address, 'root', 600)
+When(/^I check the MAC address value on "([^"]*)"$/) do |host|
+  node = get_target(host)
+  get_system_id_like = "awk -F= '/^ID_LIKE/{print $2}' /etc/os-release"
+  id_like, _local, _remote, _code = node.test_and_store_results_together(get_system_id_like, 'root', 600)
+  iface = id_like == 'debian' ? 'ens3' : 'eth0'
+  get_mac_address = "cat /sys/class/net/#{iface}/address"
+  mac_address, _local, _remote, _code = node.test_and_store_results_together(get_mac_address, 'root', 600)
```

Comment:
`run()` is note sufficient?

Reply:
I didn't wanted to change that much the previous code, but you are right, I will profit to change some of them with just run method, which is just a wrapper of that in fact.